src/pages/Login/LoginPage.tsx

```diff
@@ -1,36 +1,18 @@
 import LoginImage from '@assets/Login/LoginImage.svg'
 import TrickerLogo from '@assets/TrickerLogo'
 import TrickerTitle from '@assets/TrickerTitle'
-
 import { LoginWithButton } from '@components/LoginWithButton/LoginWithButton'
 import { NeedHelpButton } from '@components/NeedHelpButton/NeedHelpButton'
-import * as service from '@data-provider/service'
-import React from 'react'
+import { useVerifyToken } from '@data-provider/query'
 import { useNavigate, useSearchParams } from 'react-router-dom'
-import { useAppDispatch } from '@redux/hooks'
 
 const LoginPage = (): JSX.Element => {
   const [searchParams] = useSearchParams()
   const navigate = useNavigate()
-  const dispatch = useAppDispatch()
 
-  React.useEffect((): void => {
-    const verifyToken = (): void => {
-      const code = searchParams.get('code')
-      if (code !== null) {
-        service
-          .verifyToken(code)
-          .then(() => {
-            navigate('/')
-            return null
-          })
-          .catch((error) => {
-            console.error('Error verifying token:', error)
-          })
-      }
-    }
-    verifyToken()
-  }, [searchParams, navigate, dispatch])
+  const code = searchParams.get('code')
+  const { data } = useVerifyToken(code || '')
+  if (data) navigate('/')

```

Comment:
If it is done in that way, you will have the following warning: Cannot update a component (`RouterProvider`) while rendering a different component (`LoginPage`). To locate the bad setState() call inside `LoginPage`
To fix this issue, you can use the useEffect hook to perform the token verification when the component mounts. Refactor your code:
```
useEffect(() => {
    if (data) {
      navigate('/');
    }
  }, [data, navigate]);
```